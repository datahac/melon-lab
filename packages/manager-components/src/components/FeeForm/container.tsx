import { withFormik } from 'formik';
import { compose, withHandlers } from 'recompose';
import * as Yup from 'yup';

import FeeForm from './index';

const calculateTotal = (props, field, value) => {
  const { values } = props;
};

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.initialValues }),
  enableReinitialize: true,
  handleSubmit: (values, form) => {
    if (form.props.onSubmit) {
      form.props.onSubmit(values, 'passwordForm');
    }
  },
});

const withFormHandler = compose(
  withHandlers({
    onChange: props => (values, event) => {
      props.setFieldValue(event.target.name, values.value);
      calculateTotal(props, event.target.name, values.value);
    },
  }),
);

export default compose(
  withFormValidation,
  withFormHandler,
)(FeeForm);
