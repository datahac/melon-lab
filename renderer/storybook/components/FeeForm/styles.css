.fee-form__buttons {
  margin-top: var(--space-m);
  display: flex;
  width: 100%;
  justify-content: space-between;
}

.fee-form__button {
  flex: 0 0 40%;
}

.fee-form__input {
  margin: 0 0 var(--space-m) 0;
}
