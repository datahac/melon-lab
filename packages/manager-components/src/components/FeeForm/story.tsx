import { storiesOf } from '@storybook/react';
import React from 'react';
import FeeForm from './container';

const initialProps = {
  initialValues: {
    gasPrice: '20.000',
  },
  fees: [
    {
      description: 'Estimated max gas cost',
      gasLimit: 30000,
      gasPrice: '20',
      gasTotal: '0.0006',
    },
    {
      description: 'Estimated max gas cost',
      gasLimit: 30000,
      gasPrice: '20',
      gasTotal: '0.0006',
    },
  ],
};

storiesOf('Components|Fees Form', module).add('Default', () => {
  return <FeeForm {...initialProps} />;
});
