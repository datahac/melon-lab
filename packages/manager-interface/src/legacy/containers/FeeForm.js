import * as R from 'ramda';
import { connect } from 'react-redux';
import FeeForm from '@melonproject/manager-components/components/FeeForm/container';
import { actions } from '../actions/modal';

const mapStateToProps = state => {
  return {
    initialValues: {
      gasPrice: R.pathOr(20, ['form', 'modal', 'values', 'gasPrice'], state),
    },
    fees: state.modal.fees,
  };
};

const mapDispatchToProps = dispatch => ({
  onSubmit: values => {
    dispatch(actions.confirmed(values.gasPrice * 10 ** 9));
  },
  onCancel: () => {
    dispatch(actions.cancel());
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(FeeForm);
