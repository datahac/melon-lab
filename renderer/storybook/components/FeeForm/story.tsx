import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import FeeForm from './index';

const data = {
  values: {
    gasPrice: 2000000000,
  },
  touched: {
    gasPrice: null,
  },
  errors: {
    gasPrice: null,
  },
  gasLimit: 30000,
  handleSubmit: action('handleSubmit'),
  handleChange: action('handleChange'),
  handleBlur: action('handleBlur'),
  onCancel: action('onCancel'),
};

storiesOf('Components|Fee Form', module).add('Default', () => {
  return <FeeForm {...data} />;
});
