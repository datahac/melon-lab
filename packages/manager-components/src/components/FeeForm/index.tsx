import React, { StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';
import {
  CellBody,
  CellHead,
  Row,
  Table,
  TableBody,
  TableHead,
} from '~/blocks/Table';
import { add } from '../../utils/functionalBigNumber';

import styles from './styles.css';

interface FormValues {
  gasPrice: number;
}

export interface FeeFormProps {
  values: FormValues;
  handleSubmit: () => void;
  handleReset: () => void;
  onCancel: () => void;
  handleBlur?: () => void;
  onChange?: () => void;
  touched?: any;
  errors?: any;
  gasPrice;
  fees;
}

export const FeeForm: StatelessComponent<FeeFormProps> = ({
  values,
  handleBlur,
  onChange,
  touched,
  errors,
  handleSubmit,
  onCancel,
  handleReset,
  gasPrice,
  fees,
}) => {
  const handleCancel = () => {
    handleReset();
    onCancel();
  };

  return (
    <div className="fee-form">
      <style jsx>{styles}</style>
      <Form>
        <div className="fee-form__input">
          <Input
            value={values.gasPrice}
            type="number"
            label="Gas price"
            name="gasPrice"
            insideLabel="true"
            onChange={onChange}
            onBlur={handleBlur}
            required={true}
            formatNumber={true}
            error={touched.gasPrice && errors.gasPrice}
          />
        </div>

        {fees && (
          <div>
            <Table>
              <TableHead>
                <Row isHead>
                  <CellHead>Description</CellHead>
                  <CellHead>Gas Limit</CellHead>
                  <CellHead>Total</CellHead>
                </Row>
              </TableHead>
              <TableBody>
                {fees.map((entry, i) => (
                  <Row key={`fee-${i}`}>
                    <CellBody>{entry.description}</CellBody>
                    <CellBody>{entry.gasLimit}</CellBody>
                    <CellBody>
                      Ξ {(values.gasPrice * entry.gasLimit) / 10 ** 9}
                    </CellBody>
                  </Row>
                ))}

                <Row>
                  <CellHead />
                  <CellHead />
                  <CellHead>
                    Ξ{' '}
                    {add(
                      ...fees.map(
                        e => (e.gasLimit * values.gasPrice) / 10 ** 9,
                      ),
                    )}
                  </CellHead>
                </Row>
              </TableBody>
            </Table>
            <p>
              If you do not change the gas price field, the default gas price
              will be used. If you wish to set the gas price according to
              network conditions, please refer to{' '}
              <a href="https://ethgasstation.info/" target="_blank">
                Eth Gas Station.
              </a>
            </p>
          </div>
        )}

        <div className="fee-form__buttons">
          <div className="fee-form__button">
            <Button type="button" style="secondary" onClick={handleCancel}>
              Cancel
            </Button>
          </div>
          <div className="fee-form__button">
            <Button type="submit" onClick={handleSubmit}>
              Confirm
            </Button>
          </div>
        </div>
      </Form>
    </div>
  );
};

export default FeeForm;
