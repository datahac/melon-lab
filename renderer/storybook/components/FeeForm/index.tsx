import React, { StatelessComponent } from 'react';
import Input from '~/blocks/Input';
import {
  CellBody,
  CellHead,
  Row,
  Table,
  TableBody,
  TableHead,
} from '~/blocks/Table';

import styles from './styles.css';

interface FormValues {
  gasPrice: number;
}

interface FormErrors {
  gasPrice: string;
}

export interface FeeFormProps {
  error?: Error;
  errors: FormErrors;
  handleBlur: () => void;
  handleChange: () => void;
  touched?: any;
  values: FormValues;
  text?: string;
  gasLimit: number;
}

export const FeeForm: StatelessComponent<FeeFormProps> = ({
  error,
  errors,
  handleBlur,
  handleChange,
  touched,
  values,
  text,
  gasLimit,
}) => {
  const calcEntryTotal = (gasLimit: number) => {
    return gasLimit && values.gasPrice && values.gasPrice * gasLimit;
  };
  const total = gasLimit * values.gasPrice;

  return (
    <div className="fee-form">
      <style jsx>{styles}</style>
      {error && <p>{error.message}</p>}
      {!error && text && <p>{text}</p>}
      {!error && (
        <div className="fee-form__input">
          <Input
            value={values.gasPrice.toString()}
            label="Gas price (WEI)"
            name="gasPrice"
            insideLabel
            onChange={handleChange}
            onBlur={handleBlur}
            required={true}
            formatNumber={true}
            error={touched.gasPrice && errors.gasPrice}
          />
        </div>
      )}

      {!error && gasLimit && (
        <div>
          <Table>
            <TableHead>
              <Row isHead>
                <CellHead>Description</CellHead>
                <CellHead>Gas Limit</CellHead>
                <CellHead>Total</CellHead>
              </Row>
            </TableHead>
            <TableBody>
              <Row>
                <CellBody />
                <CellBody>{gasLimit}</CellBody>
                <CellBody>Ξ {calcEntryTotal(gasLimit)}</CellBody>
              </Row>

              <Row>
                <CellHead />
                <CellHead />
                <CellHead>Ξ {total}</CellHead>
              </Row>
            </TableBody>
          </Table>
          <p>
            If you do not change the gas price field, the default gas price will
            be used. If you wish to set the gas price according to network
            conditions, please refer to{' '}
            <a href="https://ethgasstation.info/" target="_blank">
              Eth Gas Station.
            </a>
          </p>
        </div>
      )}
    </div>
  );
};

export default FeeForm;
