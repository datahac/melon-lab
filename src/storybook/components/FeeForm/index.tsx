import React, { StatelessComponent } from 'react';
import Input from '~/blocks/Input';
import { CellBody, CellHead, Row, Table, TableBody, TableHead } from '~/blocks/Table';
import Notification from '~/blocks/Notification';
import * as Tm from '@melonproject/token-math';

import styles from './styles.css';

interface FormValues {
  gasPrice: number;
}

interface FormErrors {
  gasPrice: string;
}

export interface FeeFormProps {
  error?: Error;
  errors?: FormErrors;
  handleBlur?: () => void;
  handleChange?: () => void;
  touched?: any;
  values: FormValues;
  text?: string;
  gasLimit: number;
  total: Tm.QuantityInterface;
  description?: string;
  amguInEth: Tm.QuantityInterface;
  incentiveInEth: Tm.QuantityInterface;
}

export const FeeForm: StatelessComponent<FeeFormProps> = ({
  error,
  errors,
  handleBlur,
  handleChange,
  touched,
  values,
  text,
  gasLimit,
  total,
  description,
  amguInEth,
  incentiveInEth,
}) => (
  <div className="fee-form">
    <style jsx>{styles}</style>
    {error && <Notification isError>{error.message}</Notification>}
    {!error && text && <p>{text}</p>}
    {!error && (
      <div className="fee-form__input">
        <Input
          value={values.gasPrice}
          label="Gas price (GWEI)"
          name="gasPrice"
          insideLabel={true}
          onChange={handleChange}
          onBlur={handleBlur}
          required={true}
          formatNumber={true}
          error={touched.gasPrice && errors.gasPrice}
        />
      </div>
    )}

    {!error && gasLimit && (
      <div>
        <Table>
          <TableHead>
            <Row isHead={true}>
              <CellHead>Description</CellHead>
              <CellHead />
              <CellHead textAlign="right">Cost</CellHead>
            </Row>
          </TableHead>
          <TableBody>
            <Row>
              <CellBody>{description}</CellBody>
              <CellBody>Gas limit: {gasLimit}</CellBody>
              <CellBody textAlign="right">Ξ {total && Tm.toFixed(total, 6)}</CellBody>
            </Row>
            {amguInEth && (
              <Row>
                <CellBody>Amgu</CellBody>
                <CellBody />
                <CellBody textAlign="right">Ξ {Tm.toFixed(amguInEth, 6)}</CellBody>
              </Row>
            )}
            {incentiveInEth && (
              <Row>
                <CellBody>Incentive</CellBody>
                <CellBody />
                <CellBody textAlign="right">Ξ {Tm.toFixed(incentiveInEth, 6)}</CellBody>
              </Row>
            )}
          </TableBody>
        </Table>
        <p>
          If you do not change the gas price field, the default gas price will be used. If you wish to set the gas price
          according to network conditions, please refer to{' '}
          <a href="https://ethgasstation.info/" target="_blank">
            Eth Gas Station.
          </a>
        </p>
      </div>
    )}
  </div>
);

export default FeeForm;
