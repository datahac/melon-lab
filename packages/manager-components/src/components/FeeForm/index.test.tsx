import React from 'react';
import FeeForm from './container';

const initialProps = {
  initialValues: {
    gasPrice: '20.000',
  },
};

describe('FeeForm', () => {
  const defaultElement = <FeeForm {...initialProps} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
