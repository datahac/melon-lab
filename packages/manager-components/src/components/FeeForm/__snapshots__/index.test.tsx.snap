// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeeForm should render correctly 1`] = `
<Formik
  enableReinitialize={true}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "gasPrice": "20.000",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
/>
`;
